src/tests/handlers/homework.test.ts

```diff
@@ -0,0 +1,118 @@
+                        homeworkMock.find((homework) => homework.id === homeworkId)
+                    )                             
+                }),
+                query: jest.fn((homeworkId,userEmail)=> {
+                    return Promise.resolve(
+                        homeworkSubmissionMock.filter((submission) => submission.homeworkId === homeworkId && submission.studentEmail === userEmail)
+                    )                             
+                })
+            }
+        })
+    }) 
+})
+
+jest.mock('../../helpers/validations',() => {
+    return({
+        validateToken: jest.fn((token) => {
+            return (token ? {
+                userType: userMock[0].userType,
+                userEmail: userMock[0].email
+            } : null)                  
+        })
+    })
+})
+
+describe("Homework handler", () => {
+    describe("Fail test", () => {
+        it("Should fail when user token is not given", async () => {
+            const response = await handler({
+                headers: {
+                    Authorization: null
+                }
+            })
+            
+            const body = JSON.parse(response.body)
+            expect(body.message).toEqual("Fail Authenticating");
+            expect(response.statusCode).toEqual(403);      
+        })
+
+        it("Should fail when the Homework ID is not given", async () => {
+            const response = await handler({
+                headers: {
+                    Authorization: "someGivenToken"
+                },
+                pathParameters: null ?? {
+                    homeworkId: " "
+                }
+            })
+            
+            const body = JSON.parse(response.body)
+            expect(body.message).toEqual("Homework ID is required");
+            expect(response.statusCode).toEqual(400);      
+        })
+
+        it("Should fail when the Homework ID is not found", async () => {
+            const response = await handler({
+                headers: {
+                    Authorization: "someGivenToken"
+                },
+                pathParameters: {
+                    homeworkId: "4"
+                }
+            })
+            
+            const body = JSON.parse(response.body)
+            expect(body.message).toEqual("Homework not found");
+            expect(response.statusCode).toEqual(400);      
+        })
+
+        it("Should fail when the student don't have submissions", async () => {
+            const response = await handler({
+                headers: {
+                    Authorization: "someGivenToken"
+                },
+                pathParameters: {
+                    homeworkId: "3"
+                }
+            })
+            
+            const body = JSON.parse(response.body)  
+            expect(body.data.grade).toEqual(false);         
+            expect(body.data.pendingSubmission).toEqual(true); 
+            expect(body.message).toEqual("Success");
```

Comment:
if this tests says "should fail" i don't think that the message success is a good one. Or the test name is not correctly written